### PR TITLE
Silence "Extension context invalidated" -error

### PR DESCRIPTION
**Pull request by KillariDev**
Fixes https://github.com/DarkFlorist/TheInterceptor/issues/17

### CODE REVIEW

File: extension/app/inpage/output/injected_document_start.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ function listenInContentScript() {
         }
         catch (error) {
             // CONSIDER: should we catch data clone error and then do `extensionPort.postMessage({data:JSON.parse(JSON.stringify(messageEvent.data))})`?
+            if (error instanceof Error) {
+                if (error.message?.includes('Extension context invalidated.')) {
+                    // this error happens when the extension is refreshed and the page cannot reach The Interceptor anymore
+                    return;
+                }
+            }
             console.error(error);
```
Comment (MicahZoltu) [inline extension/app/inpage/output/injected_document_start.js:25]: Should this still be `console.error`, or should we rethrow it here?

**Review by MicahZoltu — APPROVED** [review]
Approved if swallowing the error is still the correct behavior instead of rethrowing.

File: extension/app/inpage/output/injected_document_start.js (hunk 1 above)
Comment (KillariDev) [reply extension/app/inpage/output/injected_document_start.js:25]: I am not sure what other errors could occur. Maybe rethrow, and then it would be correctly logged as unhandled exception

Comment (MicahZoltu) [reply extension/app/inpage/output/injected_document_start.js:25]: My general exception handling policy is to handle exceptions anywhere you can fully recover regardless of failure.  This leads to my exception blocks always being incredibly generic, and I often have just a few in the whole application.  The idea is that you never know in advance what kinds of failures you will have, so it is best to handle all failures in as few places as possible, where you can spend a lot of time thinking about exactly how to deal with a particular situation.

In this case, we have one specific error that we intentionally want to ignore, but for all other possible errors we should fall back to our generalized exception handling (i.e., rethtrow).

TL;DR: I agree.
